### ISSUE TITLE
/RTC compiler checks failures don't break into debugger

### ISSUE THREAD

**Issue by giuntnic** [microsoft/vscode#140380]
- Running VSCode version 1.63.2
- Windows version 20H2 19042.1415
- Extension installed: Microsoft's C/C++ extension
I've attached a minimal app that reproduces this issue along with the cmake file to build it in [src.zip](https://github.com/microsoft/vscode/files/7836264/src.zip). This sample can be built:

1. 'cmake -S . -B build/'
2. 'cmake --build build/ --config Debug'
3. launch.json must be configured to enable running and debugging the build executable through VSCode
4. In the Breakpoints UI toggle 'All Exceptions'
5. Run the sample through VSCode

Result
```
Run-Time Check Failure #1 - A cast to a smaller data type has caused a loss of data.  If this was intentional, you should mask the source of the cast with the appropriate bitmask.  For example:  
	char c = (i & 0xFF);
Changing the code in this way will not affect the quality of the resulting optimized code.
```
is printed to the output console.

Expected
IIn addition to the above output, the debugger should break when the RTC failure occurs per documentation [here](https://docs.microsoft.com/en-us/visualstudio/debugger/how-to-use-native-run-time-checks?view=vs-2022). Specifically:
> When you debug a program that has run-time checks enabled, the default action is for the program to stop and break to the debugger when a run-time error occurs.

TLDR:
The following code snippet:
```
        const size_t violate_rtcc = UCHAR_MAX + 1 + rand();
        const unsigned char bad_assignment = violate_rtcc;
```
Compiled with /RTCcs only prints a failure message to console and doesn't break into the debugger.
